src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -144,6 +159,76 @@ def convert_streamlit_to_v2_data(streamlit: Streamlit):
     return data
 
 
+def convert_native_app_to_v2_data(
+    project_root, native_app: NativeApp
+) -> Dict[str, Any]:
+    def _make_meta(obj: Application | Package):
+        meta = {}
+        if obj.role:
+            meta["role"] = obj.role
+        if obj.warehouse:
+            meta["warehouse"] = obj.warehouse
+        if obj.post_deploy:
+            meta["post_deploy"] = obj.post_deploy
+        return meta
+
+    def _find_manifest():
+        # We don't know which file in the project directory is the actual manifest,
+        # and we can't iterate through the artifacts property since the src can contain
+        # glob patterns. The simplest solution is to bundle the app and find the
+        # manifest file from the resultant BundleMap, since the bundle process ensures
+        # that only a single source path can map to the corresponding destination path
+        bundle_map = build_bundle(
+            project_root, Path(native_app.deploy_root), native_app.artifacts
+        )
```

Comment:
The default behaviour is that the `build_bundle` will fail here if there are no artifacts specified. Would that make sense work for this conversion to v2 as well? Or should the conversion happen with the artifact list empty? If the latter, then you might want to catch that error here :)

Reply:
the manifest field is mandatory in v2, i'll make this a user error in a followup